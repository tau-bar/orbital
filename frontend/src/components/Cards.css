.cards {
  width:100%;
  background-image: url('https://www.solidbackgrounds.com/images/950x350/950x350-black-solid-color-background.jpg');
}

h1 {
  text-align: center;
  color:lavender;
}

.cards__container {
  display: flex;
  flex-flow: column;
  align-items: center;
  max-width: 2000px;
  width: 100%;
  margin: 0 auto;
}

.cards__wrapper {
  position: relative;
}

.cards__items {
  margin-bottom: 24px;
}

.cards__item {
  display: flex;
  flex: 3;
  margin: 0 3rem;
}


.cards__item__pic-wrap {
  position: relative;
  overflow: hidden;
  background: black;
  border-radius: 1000px;
}

.fade-img {
  animation-name: fade-img;
  animation-duration: 2s;
}

.cards__item__pic-wrap:after {
  content: attr(data-category);
  position: absolute;
  bottom: 0;
  margin-left: 10px;
  padding: 6px 6px;
  max-width: calc((100%) - 40px);
  font-size: 12px;
  font-weight: 500;
  color: #fff;
  box-sizing: border-box;
  opacity:0;
}

.cards__item__pic-wrap:hover:after {
  content: attr(data-category);
  position: absolute;
  bottom: 130px;
  right: 50px;
  margin-left: 0px;
  margin-right: 5px;
  padding: 20px 10px;
  max-width: calc((100%) - 60px);
  font-size: 20px;
  font-weight: 700;
  color:white;
  box-sizing: border-box;
  opacity:1;
}

.cards__item__img {
  display: block;
  height: auto;
  max-height: 250px;
  width: auto;
  max-width: 250px;
  transition: all 0.2s linear;
  border-radius: 20%;
  margin-bottom: 20%;
}

.cards__item__img:hover {
  transform: scale(1.1);
}

.cards__item__text {
  color: #252e48;
  font-size: 40px;
  line-height: 24px;
  margin-top: 10px;
}

@media only screen and (min-width: 1200px) {
  .content__blog__container {
    width: 84%;
  }
}

@media only screen and (min-width: 1024px) {
  .cards__items {
    display: flex;
  }
}

@media only screen and (max-width: 1024px) {
  .cards__item {
    margin-bottom: 2rem;
  }
}