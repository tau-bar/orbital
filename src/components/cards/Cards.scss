.cards {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.cards-title {
  color: black;
  padding-top: 30px;
}

.cards-container {
  width: 60%;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
}


.card-item {
  width: 200px;
  max-width: 220px;
  height: 200px;
  flex: 3 3 auto;
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: 20px;
  background-color: black;
  margin: 0 15px 15px;
  overflow: hidden;

  &:hover {
    cursor: pointer;
    
    
    & .background-image {
      transform: scale(1.1);
      transition: transform 1s cubic-bezier(0.25, 0.45, 0.45, 0.95);
    }
  }

  .background-image {
    width : 100%;
    height: 100%;
    background-position: center;
    background-size: cover;
  }

  .content {
    height: 200px;
    width: 220px;
    padding: 0 0;
    display: flex;
    flex-direction: column;
    align-items: flex-end;
    justify-content: end;
    color: white;
    opacity: 0;
    position: absolute;
    

    &:hover {
      opacity: 1;
    }

    .subtitle {
      font-size: 20px;
      font-weight: bold;
      padding-right: 20px;
    }
  }
}

@media screen and (max-width: 800px){
  .menu-item{
    height: 200px;

    &.large{
      height: 200px;
    }

  .content{
    padding:10px;
  }

  }
}