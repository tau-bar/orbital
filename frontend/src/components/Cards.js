import React from 'react';
import './Cards.scss';
import CardItem from './CardItem';
import { withRouter } from 'react-router-dom';

const Cards = (props) => {
  const virusData = {
    coronavirus: {
      src:'/images/corona2.jpeg'
    },
    flavivirus: {
      src:'/images/fla.jpeg'
    },
    mobillivirus: {
      src:'/images/morbi.jpeg'
    },
    yersenia: {
      src:'/images/yersenia3.jpeg'
    },
    ebolavirus: {
      src:'/images/ebola2.jpeg'
    },
    orthopoxvirus : {
      src:'/images/ortho4.jpeg'
    },
    betacoronavirus: {
      src:'/images/corona2.jpeg'
    }
  }

  return (
    <div className='cards'>
      <h1>Understanding Viruses</h1>
      <div className='cards__container'>

          <div className='cards__items'>
          {
            Object.keys(virusData).map((key) => {
              console.log(key)
              return(
                <CardItem
                src= {virusData[key].src}
                label={key.charAt(0).toUpperCase() + key.slice(1)}
                onClick={() => props.history.push(`virus/${key}`)}
              />)
            }
              )
          }
          </div>
      </div>
    </div>
  );
}

export default withRouter(Cards);