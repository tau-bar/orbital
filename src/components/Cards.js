import React from 'react';
import './Cards.scss';
import CardItem from './CardItem';
import { withRouter } from 'react-router-dom';

const Cards = (props) => {
  const virusData = {
    coronavirus: {

      src:'/images/coronavirus.png'
    },
    flavivirus: {
      src:'/images/flavivirus.png'
    },
    mobillivirus: {
      src:'/images/morbillivirus.png'
    },
    yersenia: {

      src:'/images/yersinia.png'
    },
    ebolavirus: {
      src:'/images/ebola.jpeg'
    },
    orthopoxvirus : {

      src:'/images/orthopox.png'
    },
   
  }

  return (
    <div className='cards'>
      <h1 className = 'cards-title'>Understanding Viruses</h1>
      <div className='cards__container'>

          <div className='cards__items'>
          {
            Object.keys(virusData).map((key) => {
              return(
                <CardItem
                src= {virusData[key].src}
                label={key.charAt(0).toUpperCase() + key.slice(1)}
                onClick={() => props.history.push(`/virus/model/${key}`)}
              />)
            }
              )
          }
          </div>
      </div>
    </div>
  );
}

export default withRouter(Cards);