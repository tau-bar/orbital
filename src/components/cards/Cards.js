import React, { useContext } from 'react';
import './Cards.scss';
import '../../pages/virus-page/virus-page.styles.scss';
import CardItem from './CardItem';
import { withRouter } from 'react-router-dom';
import viruses from '../../assets/images/viruses.jpg'
import { HamburgerContext } from '../../context/HamburgerProvider';


export const virusData = {
  coronavirus: {
    src:'/images/coronavirus.png'

  },
  flavivirus: {
    src:'/images/flavivirus.png'
  },
  mobillivirus: {
    src:'/images/morbillivirus.png'
  },
  yersinia: {

    src:'/images/yersinia.png'
  },
  ebolavirus: {
    src:'/images/ebola.jpeg'
  },
  orthopoxvirus : {
    src:'/images/orthopox.png'
  },   
}

const Cards = (props) => {
  const setIsActive = useContext(HamburgerContext)[1];
  return (
    <div 
    data-testId = "cards"
    className='cards'
    style={{ backgroundImage: `url(${viruses})` }}>
      <h1 className = 'cards-title'>Understanding Viruses</h1>
      <p className = 'virus-page-text' style = {{color: 'white'}}>Click on a card to view the interactive virus model!</p>
      <div className='cards-container'>
          {
            Object.keys(virusData).map((key, index) => {
              return(
                <CardItem
                key = {`virusCards${index}`}
                src= {virusData[key].src}
                label={key.charAt(0).toUpperCase() + key.slice(1)}
                onClick={() => {
                  props.history.push({
                  pathname: `/virus/model/${key}`,
                  state: {
                    key: key,
                  }})
                  setIsActive(false);
                }}
              />)
            }
              )
          }
        </div>
    </div>
  );
}

export default withRouter(Cards);