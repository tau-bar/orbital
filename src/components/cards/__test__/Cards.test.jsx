import React from 'react';
import Cards from '../Cards';
import { BrowserRouter as Router } from 'react-router-dom';
import { cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

afterEach(() => {
    cleanup();
}) 

it("cards matches snapshot", () => {
    const cards = renderer.create(
        <Router>
            <Cards/>
        </Router>
        ).toJSON();
    expect(cards).toMatchSnapshot();
})

