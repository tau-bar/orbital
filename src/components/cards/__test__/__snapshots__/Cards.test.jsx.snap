// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cards matches snapshot 1`] = `
<div
  className="cards"
  data-testId="cards"
  style={
    Object {
      "backgroundImage": "url(viruses.jpg)",
    }
  }
>
  <h1
    className="cards-title"
  >
    Understanding Viruses
  </h1>
  <p
    className="virus-page-text"
    style={
      Object {
        "color": "white",
      }
    }
  >
    Click on a card to view the interactive virus model!
  </p>
  <div
    className="cards-container"
  >
    <div
      className="card-item"
      onClick={[Function]}
    >
      <div
        className="background-image"
        style={
          Object {
            "backgroundImage": "url(/images/coronavirus.png)",
          }
        }
      />
      <div
        className="content"
      >
        <span
          className="subtitle"
        >
          Coronavirus
        </span>
      </div>
    </div>
    <div
      className="card-item"
      onClick={[Function]}
    >
      <div
        className="background-image"
        style={
          Object {
            "backgroundImage": "url(/images/flavivirus.png)",
          }
        }
      />
      <div
        className="content"
      >
        <span
          className="subtitle"
        >
          Flavivirus
        </span>
      </div>
    </div>
    <div
      className="card-item"
      onClick={[Function]}
    >
      <div
        className="background-image"
        style={
          Object {
            "backgroundImage": "url(/images/morbillivirus.png)",
          }
        }
      />
      <div
        className="content"
      >
        <span
          className="subtitle"
        >
          Mobillivirus
        </span>
      </div>
    </div>
    <div
      className="card-item"
      onClick={[Function]}
    >
      <div
        className="background-image"
        style={
          Object {
            "backgroundImage": "url(/images/yersinia.png)",
          }
        }
      />
      <div
        className="content"
      >
        <span
          className="subtitle"
        >
          Yersenia
        </span>
      </div>
    </div>
    <div
      className="card-item"
      onClick={[Function]}
    >
      <div
        className="background-image"
        style={
          Object {
            "backgroundImage": "url(/images/ebola.jpeg)",
          }
        }
      />
      <div
        className="content"
      >
        <span
          className="subtitle"
        >
          Ebolavirus
        </span>
      </div>
    </div>
    <div
      className="card-item"
      onClick={[Function]}
    >
      <div
        className="background-image"
        style={
          Object {
            "backgroundImage": "url(/images/orthopox.png)",
          }
        }
      />
      <div
        className="content"
      >
        <span
          className="subtitle"
        >
          Orthopoxvirus
        </span>
      </div>
    </div>
  </div>
</div>
`;
